crates/provider/src/provider/helper.rs

```diff
@@ -0,0 +1,30 @@
+use crate::{Provider, RootProvider};
+
+use alloy_network::Network;
+use std::sync::Arc;
+
+/// A wrapper struct around Arc dyn provider
+pub struct WrappedProvider<'a, N>(Arc<dyn Provider<N> + 'a>);
+
+impl<'a, N: Network> WrappedProvider<'a, N> {
+    pub fn new<P>(provider: &'a P) -> Self
+    where
+        P: Provider<N> + 'a,
+    {
+        Self(Arc::new(provider))
+    }
+}
+
+impl<N: Network> Provider<N> for WrappedProvider<'_, N> {
+    fn root(&self) -> &RootProvider<N> {
+        self.0.root()
+    }
+}
```

Comment:
I assume we need override more functions because the wrapped provider could use non default fns

so this needs to delegate all fns

Reply:
delegated all the fns except `get_filter_changes` and `raw_request`, as they required `Sized`

we already had their dyn version